### PR TITLE
Clarify session joining prerequisites

### LINKED ISSUE #1968: Update documentation to include prerequisites for session sharing
**What happened**:
After building from the master branch and installing teleport I am not able to share a session with another user due to an error.
`Original Error: *trace.AccessDeniedError can't switch users from paul to moritz for session b40d598e-5d03-11e8-a873-000c292bd3bb`

**What you expected to happen**:
After adding the user `paul` to teleport I was expecting being able to share the session with him.

**How to reproduce it (as minimally and precisely as possible)**:
Here are the steps I took when I encountered the issue.

* Build from master branch and installed it on `moritz-virtual-machine`
* As root I ran teleport without any configuration file or parameters
`root@moritz-virtual-machine:~# teleport start -d`
* I added the OS user `moritz` to teleport and verified that I could `tsh login` to the cluster as well as the WebAdmin
`tsh --insecure --proxy=10.150.4.72 login`
* After reading https://gravitational.com/teleport/docs/quickstart/#sharing-ssh-sessions I came across the info box that says 
> For this to work, both of you must have proper user mappings allowing you access db under the same OS user.  

I added the user `paul` on `moritz-virtual-machine` via `adduser`; as this was my understanding of this info box. But I am not sure if I understood this right. It does not seem logical to me to first add individual linux users to the node where I want to share a session with a teleport user under the same name.

* On a second VM `debian` I have the local user `paul`
* I installed the `tsh` binary that I built earlier and successfully logged into the cluser
```
paul@debian:/root$ tsh --insecure --proxy=moritz-virtual-machine login
WARNING: You are using insecure connection to SSH proxy https://moritz-virtual-machine:3080
Enter password for Teleport user paul:
Enter your OTP token:
066455
WARNING: You are using insecure connection to SSH proxy https://moritz-virtual-machine:3080

You are now logged in
paul@debian:/root$
```
Then I ssh`ed into the cluster and printed the session for my user `moritz`, which is local to the cluster, to join there.
```
paul@debian:~$ tsh --insecure --proxy=moritz-virtual-machine login
WARNING: You are using insecure connection to SSH proxy https://moritz-virtual-machine:3080
Enter password for Teleport user paul:
Enter your OTP token:
174116
WARNING: You are using insecure connection to SSH proxy https://moritz-virtual-machine:3080

You are now logged in
paul@debian:~$ tsh --insecure --proxy=moritz-virtual-machine ssh paul@moritz-virtual-machine
paul@moritz-virtual-machine:~$ teleport status
User ID     : paul, logged in as paul from 10.150.4.69 53524 3022
Cluster Name: moritz-virtual-machine
Host UUID   : 0f0c1d91-9f40-4d69-be42-fd93656bcc48
Session ID  : 21978b90-5d10-11e8-b3e8-000c292bd3bb
Session URL : https://moritz-virtual-machine:3080/web/cluster/moritz-virtual-machine/node/0f0c1d91-9f40-4d69-be42-fd93656bcc48/paul/21978b90-5d10-11e8-b3e8-000c292bd3bb
paul@moritz-virtual-machine:~$ the connection was closed on the remote side on  21 May 18 18:01 CEST
paul@debian:~$
```
From the VM `moritz-virtual-machine` where the teleport cluster is running I wanted to join the session.
```
moritz@moritz-virtual-machine:~$ tsh --insecure --proxy=moritz-virtual-machine login
WARNING: You are using insecure connection to SSH proxy https://moritz-virtual-machine:3080
Enter password for Teleport user moritz:
Enter your OTP token:
518823
WARNING: You are using insecure connection to SSH proxy https://moritz-virtual-machine:3080

You are now logged in
moritz@moritz-virtual-machine:~$ tsh --insecure --proxy=moritz-virtual-machine join 21978b90-5d10-11e8-b3e8-000c292bd3bb

ERROR REPORT:
Original Error: *trace.AccessDeniedError can't switch users from paul to moritz for session 21978b90-5d10-11e8-b3e8-000c292bd3bb
Stack Trace:
	/home/moritz/go/src/github.com/gravitational/teleport/lib/srv/sess.go:906 github.com/gravitational/teleport/lib/srv.(*session).addParty
	/home/moritz/go/src/github.com/gravitational/teleport/lib/srv/sess.go:948 github.com/gravitational/teleport/lib/srv.(*session).join
	/home/moritz/go/src/github.com/gravitational/teleport/lib/srv/sess.go:154 github.com/gravitational/teleport/lib/srv.(*SessionRegistry).OpenSession
	/home/moritz/go/src/github.com/gravitational/teleport/lib/srv/termhandlers.go:175 github.com/gravitational/teleport/lib/srv.(*TermHandlers).HandleShell
	/home/moritz/go/src/github.com/gravitational/teleport/lib/srv/regular/sshserver.go:938 github.com/gravitational/teleport/lib/srv/regular.(*Server).dispatch
	/home/moritz/go/src/github.com/gravitational/teleport/lib/srv/regular/sshserver.go:885 github.com/gravitational/teleport/lib/srv/regular.(*Server).handleSessionRequests
	/usr/local/go/src/runtime/asm_amd64.s:2362 runtime.goexit
User Message: can't switch users from paul to moritz for session 21978b90-5d10-11e8-b3e8-000c292bd3bb
error: ssh: could not start shell
moritz@moritz-virtual-machine:~$ 
```

The debug log for the `teleport` process it self I could gather the following log lines:
```
DEBU [PROXY]     conn(127.0.0.1:50850->127.0.1.1:3023, user=moritz) auth attempt fingerprint:ssh-rsa-cert-v01@openssh.com SHA256:2NLeWM2PoOtRI16bSXZ+91PlvIL8MsNV2YUXbr/ikIM local:127.0.1.1:3023 remote:127.0.0.1:50850 user:moritz srv/authhandlers.go:124
DEBU [PROXY]     conn(127.0.0.1:50850->127.0.1.1:3023, user=moritz) auth attempt with key ssh-rsa-cert-v01@openssh.com SHA256:2NLeWM2PoOtRI16bSXZ+91PlvIL8MsNV2YUXbr/ikIM, &ssh.Certificate{Nonce:[]uint8{0x1e, 0xb4, 0x74, 0x2, 0xf0, 0xb0, 0x87, 0x1, 0x1f, 0xd1, 0x95, 0xfd, 0xba, 0xcf, 0xdd, 0x15, 0x8e, 0x92, 0x99, 0x48, 0x1a, 0x3f, 0x8, 0x38, 0xd, 0xbd, 0x7d, 0x28, 0x68, 0x20, 0x3, 0x4}, Key:(*ssh.rsaPublicKey)(0xc4201c8200), Serial:0x0, CertType:0x1, KeyId:"moritz", ValidPrincipals:[]string{"moritz"}, ValidAfter:0x0, ValidBefore:0x5b039612, Permissions:ssh.Permissions{CriticalOptions:map[string]string{}, Extensions:map[string]string{"permit-agent-forwarding":"", "permit-port-forwarding":"", "permit-pty":"", "teleport-roles":"{\"version\":\"v1\",\"roles\":[\"admin\"]}"}}, Reserved:[]uint8{}, SignatureKey:(*ssh.rsaPublicKey)(0xc4201c8240), Signature:(*ssh.Signature)(0xc420604720)} fingerprint:ssh-rsa-cert-v01@openssh.com SHA256:2NLeWM2PoOtRI16bSXZ+91PlvIL8MsNV2YUXbr/ikIM local:127.0.1.1:3023 remote:127.0.0.1:50850 user:moritz srv/authhandlers.go:124
DEBU [PROXY]     Successfully authenticated fingerprint:ssh-rsa-cert-v01@openssh.com SHA256:2NLeWM2PoOtRI16bSXZ+91PlvIL8MsNV2YUXbr/ikIM local:127.0.1.1:3023 remote:127.0.0.1:50850 user:moritz srv/authhandlers.go:180
DEBU [SSH:PROXY] incoming connection 127.0.0.1:50850 -> 127.0.1.1:3023 vesion: SSH-2.0-Go sshutils/server.go:390
DEBU [PROXY]     [SSH] ssh.dispatch(req=subsystem, wantReply=true) id:4 local:127.0.1.1:3023 login:moritz remote:127.0.0.1:50850 teleportUser:moritz regular/sshserver.go:907
DEBU [PROXY]     [SSH] subsystem request: proxySites() id:4 local:127.0.1.1:3023 login:moritz remote:127.0.0.1:50850 teleportUser:moritz regular/sshserver.go:1028
DEBU             proxysites.start(ServerContext(127.0.0.1:50850->127.0.1.1:3023, user=moritz, id=4)) regular/sites.go:53
DEBU             [SSH] proxySites() finished with result: <nil> regular/sshserver.go:1042
DEBU [PROXY]     [SSH] close session request: <nil> id:4 local:127.0.1.1:3023 login:moritz remote:127.0.0.1:50850 teleportUser:moritz regular/sshserver.go:878
DEBU [PROXY]     [SSH] ssh.dispatch(req=subsystem, wantReply=true) id:5 local:127.0.1.1:3023 login:moritz remote:127.0.0.1:50850 teleportUser:moritz regular/sshserver.go:907
DEBU             parse_proxy_subsys("proxy:@moritz-virtual-machine") regular/proxy.go:74
DEBU [PROXY]     [SSH] subsystem request: proxySubsys(cluster=default/moritz-virtual-machine, host=, port=) id:5 local:127.0.1.1:3023 login:moritz remote:127.0.0.1:50850 teleportUser:moritz regular/sshserver.go:1028
DEBU [SUBSYSTEM] Starting subsystem trace.fields:map[src:127.0.0.1:50850 dst:127.0.1.1:3023] regular/proxy.go:151
INFO [SUBSYSTEM] Connected to auth server: 10.150.4.72:3025 trace.fields:map[src:127.0.0.1:50850 dst:127.0.1.1:3023] regular/proxy.go:198
DEBU [PRESENCE]  GetServers(1) in 349.108µs local/presence.go:223
DEBU [SSH:PROXY] received out-of-band request: &{Type:recording-proxy@teleport.com WantReply:true Payload:[] ch:<nil> mux:0xc42014cb60} sshutils/server.go:403
DEBU             Global request (recording-proxy@teleport.com, true) received regular/sshserver.go:1080
DEBU             Replied to global request (recording-proxy@teleport.com, true): false regular/sshserver.go:1107
DEBU [PROXY]     [SSH] ssh.dispatch(req=subsystem, wantReply=true) id:6 local:127.0.1.1:3023 login:moritz remote:127.0.0.1:50850 teleportUser:moritz regular/sshserver.go:907
DEBU             parse_proxy_subsys("proxy:127.0.0.1:3022") regular/proxy.go:74
DEBU [PROXY]     [SSH] subsystem request: proxySubsys(cluster=default/, host=127.0.0.1, port=3022) id:6 local:127.0.1.1:3023 login:moritz remote:127.0.0.1:50850 teleportUser:moritz regular/sshserver.go:1028
DEBU [SUBSYSTEM] Starting subsystem trace.fields:map[src:127.0.0.1:50850 dst:127.0.1.1:3023] regular/proxy.go:151
DEBU [SUBSYSTEM] Cluster not specified. connecting to default='moritz-virtual-machine' trace.fields:map[src:127.0.0.1:50850 dst:127.0.1.1:3023] regular/proxy.go:181
DEBU [PRESENCE]  GetServers(1) in 446.217µs local/presence.go:223
DEBU [SUBSYSTEM] proxy connecting to host=127.0.0.1 port=3022, exact port=true trace.fields:map[src:127.0.0.1:50850 dst:127.0.1.1:3023] regular/proxy.go:230
DEBU [PROXY:SER] Dialing from 127.0.0.1:50850 to 127.0.0.1:3022 trace.fields:map[cluster:moritz-virtual-machine] reversetunnel/localsite.go:153
DEBU [NODE]      conn(127.0.0.1:50850->127.0.0.1:3022, user=moritz) auth attempt fingerprint:ssh-rsa-cert-v01@openssh.com SHA256:2NLeWM2PoOtRI16bSXZ+91PlvIL8MsNV2YUXbr/ikIM local:127.0.0.1:3022 remote:127.0.0.1:50850 user:moritz srv/authhandlers.go:124
DEBU [NODE]      conn(127.0.0.1:50850->127.0.0.1:3022, user=moritz) auth attempt with key ssh-rsa-cert-v01@openssh.com SHA256:2NLeWM2PoOtRI16bSXZ+91PlvIL8MsNV2YUXbr/ikIM, &ssh.Certificate{Nonce:[]uint8{0x1e, 0xb4, 0x74, 0x2, 0xf0, 0xb0, 0x87, 0x1, 0x1f, 0xd1, 0x95, 0xfd, 0xba, 0xcf, 0xdd, 0x15, 0x8e, 0x92, 0x99, 0x48, 0x1a, 0x3f, 0x8, 0x38, 0xd, 0xbd, 0x7d, 0x28, 0x68, 0x20, 0x3, 0x4}, Key:(*ssh.rsaPublicKey)(0xc4201d5920), Serial:0x0, CertType:0x1, KeyId:"moritz", ValidPrincipals:[]string{"moritz"}, ValidAfter:0x0, ValidBefore:0x5b039612, Permissions:ssh.Permissions{CriticalOptions:map[string]string{}, Extensions:map[string]string{"permit-agent-forwarding":"", "permit-port-forwarding":"", "permit-pty":"", "teleport-roles":"{\"version\":\"v1\",\"roles\":[\"admin\"]}"}}, Reserved:[]uint8{}, SignatureKey:(*ssh.rsaPublicKey)(0xc4201d5960), Signature:(*ssh.Signature)(0xc420404630)} fingerprint:ssh-rsa-cert-v01@openssh.com SHA256:2NLeWM2PoOtRI16bSXZ+91PlvIL8MsNV2YUXbr/ikIM local:127.0.0.1:3022 remote:127.0.0.1:50850 user:moritz srv/authhandlers.go:124
DEBU [NODE]      Successfully authenticated fingerprint:ssh-rsa-cert-v01@openssh.com SHA256:2NLeWM2PoOtRI16bSXZ+91PlvIL8MsNV2YUXbr/ikIM local:127.0.0.1:3022 remote:127.0.0.1:50850 user:moritz srv/authhandlers.go:180
DEBU [NODE]      Checking permissions for (moritz,moritz) to login to node with RBAC checks. fingerprint:ssh-rsa-cert-v01@openssh.com SHA256:2NLeWM2PoOtRI16bSXZ+91PlvIL8MsNV2YUXbr/ikIM local:127.0.0.1:3022 remote:127.0.0.1:50850 user:moritz srv/authhandlers.go:308
DEBU [SSH:NODE]  incoming connection 127.0.0.1:50850 -> 127.0.0.1:3022 vesion: SSH-2.0-Go sshutils/server.go:390
DEBU [NODE]      [SSH] ssh.dispatch(req=env, wantReply=true) id:7 local:127.0.0.1:3022 login:moritz remote:127.0.0.1:50850 teleportUser:moritz regular/sshserver.go:907
DEBU [NODE]      [SSH] ssh.dispatch(req=env, wantReply=true) id:7 local:127.0.0.1:3022 login:moritz remote:127.0.0.1:50850 teleportUser:moritz regular/sshserver.go:907
DEBU             [SSH] will join session session(id=21978b90-5d10-11e8-b3e8-000c292bd3bb, parties=1) for SSH connection 127.0.0.1:50850 srv/ctx.go:265
DEBU [NODE]      [SSH] ssh.dispatch(req=pty-req, wantReply=true) id:7 local:127.0.0.1:3022 login:moritz remote:127.0.0.1:50850 teleportUser:moritz regular/sshserver.go:907
DEBU [NODE]      Requested terminal "xterm-256color" of size {207 37} id:7 local:127.0.0.1:3022 login:moritz remote:127.0.0.1:50850 teleportUser:moritz srv/termhandlers.go:124
DEBU [SESSION:N] Sent resize event {207 37} to 10.150.4.69:53524. srv/sess.go:343
DEBU             [SSH] will join session session(id=21978b90-5d10-11e8-b3e8-000c292bd3bb, parties=1) for SSH connection 127.0.0.1:50850 srv/ctx.go:265
DEBU [NODE]      [SSH] ssh.dispatch(req=shell, wantReply=true) id:7 local:127.0.0.1:3022 login:moritz remote:127.0.0.1:50850 teleportUser:moritz regular/sshserver.go:907
INFO [NODE]      Joining existing session 21978b90-5d10-11e8-b3e8-000c292bd3bb. id:7 local:127.0.0.1:3022 login:moritz remote:127.0.0.1:50850 teleportUser:moritz srv/sess.go:150
ERRO             can't switch users from paul to moritz for session 21978b90-5d10-11e8-b3e8-000c292bd3bb regular/sshserver.go:1109
DEBU [TERM:LOCA] Closed PTY srv/term.go:242
DEBU [PROXY]     [SSH] client 127.0.0.1:50850 disconnected id:6 local:127.0.1.1:3023 login:moritz remote:127.0.0.1:50850 teleportUser:moritz regular/sshserver.go:883
DEBU             [SSH] proxySubsys(cluster=default/, host=127.0.0.1, port=3022) finished with result: read tcp 127.0.0.1:58160->127.0.0.1:3022: use of closed network connection regular/sshserver.go:1042
DEBU             [SSH] proxySubsys(cluster=default/moritz-virtual-machine, host=, port=) finished with result: read tcp 10.150.4.72:35262->10.150.4.72:3025: use of closed network connection regular/sshserver.go:1042
DEBU [SSH:NODE]  closed connection 127.0.0.1:50850 sshutils/server.go:392
DEBU [PROXY]     [SSH] client 127.0.0.1:50850 disconnected id:5 local:127.0.1.1:3023 login:moritz remote:127.0.0.1:50850 teleportUser:moritz regular/sshserver.go:883
DEBU [SSH:PROXY] closed connection 127.0.0.1:50850 sshutils/server.go:392
DEBU [BACKEND:D] AcquireLock(sessions21978b90-5d10-11e8-b3e8-000c292bd3bb) dir:/var/lib/teleport/backend dir/impl.go:393
DEBU [BACKEND:D] ReleaseLock(sessions21978b90-5d10-11e8-b3e8-000c292bd3bb) dir:/var/lib/teleport/backend dir/impl.go:423
DEBU [BACKEND:D] AcquireLock(sessions21978b90-5d10-11e8-b3e8-000c292bd3bb) dir:/var/lib/teleport/backend dir/impl.go:393
DEBU [BACKEND:D] ReleaseLock(sessions21978b90-5d10-11e8-b3e8-000c292bd3bb) dir:/var/lib/teleport/backend dir/impl.go:423
INFO [PROC:1]    Got signal "interrupt", exiting immediately. service/signals.go:74
DEBU [PROC:1]    Broadcasting event. event:TeleportExit service/supervisor.go:288
```

**Environment**:

- Teleport version (use `teleport version`):
`Teleport v2.6.0-rc.1 git:v2.6.0-rc.1-0-g08633daa`
- Tsh version (use `tsh version`):
`Teleport v2.6.0-rc.1 git:v2.6.0-rc.1-0-g08633daa`
- OS (e.g. from /etc/os-release):
```
IP: 10.150.4.72
moritz@moritz-virtual-machine:~$ cat /etc/os-release 
NAME="Ubuntu"
VERSION="17.10 (Artful Aardvark)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 17.10"
VERSION_ID="17.10"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=artful
UBUNTU_CODENAME=artful
moritz@moritz-virtual-machine:~$ 
```
and
```
IP: 10.150.4.69
paul@debian:~$ cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 9 (stretch)"
NAME="Debian GNU/Linux"
VERSION_ID="9"
VERSION="9 (stretch)"
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
paul@debian:~$
```

### PR DESCRIPTION
**Pull request by zmb3**
Fixes #1968

### CODE REVIEW

File: docs/pages/server-access/guides/tsh.mdx [hunk 1]
```diff
@@ -627,6 +627,11 @@ other user can join you through their terminal by typing:
 $ tsh join <session_ID>
 ```
 
+The Teleport user joining the session must have access to the same SSH principal
+used to start the session. In this case, Teleport user `joe` is logged in to the
+SSH node as Unix user `joe`, so only Teleport users with a role allowing login `joe`
```
Comment (ptgott) [inline docs/pages/server-access/guides/tsh.mdx:632]:
```suggestion
SSH Node as Unix user `joe`, so only Teleport users with a role allowing login `joe`
```

**Review by ptgott — APPROVED** [review]

**Review by russjones — APPROVED** [review]

**Comment by xacrimon** [comment]
Hey @zmb3 this isn't true as of v10. As of v10, you don't need to specify the login flag when joining and you do not need principal access as per the loosening of join restrictions we performed in v10, this is the case in v9 and older though.

**Comment by xacrimon** [comment]
You do however need a `join_sessions` policy allowing you to join, and nothing more.

**Comment by zmb3** [comment]
Thanks @xacrimon - I've canceled the v10 backport but will get this merged to v8 and v9. As discussed, can you open a change to fix master (along with the v10 backport)?

**Pull request merged**